source/falaise/snemo/services/dead_cells.h

```diff
@@ -0,0 +1,128 @@
+//! \file falaise/snemo/services/dead_cells.h
+//! \brief Tracker Dead Cells service in SuperNEMO
+//
+// more doxygen to come...
+//
+// status  description
+// ---------------------------------------------------------------------------
+// 0       good working cell            isAGoodCell==true
+// 1       dead cells with dead anode   isADeadCell==true and isABadCell==true
+// 2       cathode-to-ground short      isABadCell==true
+// 3       cathode-to-cathode short     isABadCell==true
+// 4       other                        isABadCell==true
+
+
+#ifndef SNEMO_DEAD_CELLS_SVC_HH
+#define SNEMO_DEAD_CELLS_SVC_HH
+
+#include <iostream>
+#include <fstream>
+
+#include "bayeux/datatools/base_service.h"
+#include "falaise/snemo/services/service_traits.h"
+#include "falaise/bounded_int.h"
+
+namespace snemo {
+
+  // data member for the possible statuses of a cell  
+  enum status {
+    good = 0,            // good working cell
+    dead = 1,            // dead cells with dead anode
+    cathode_ground = 2,  // cathode-to-ground short
+    cathode_cathode = 3, // cathode-to-cathode short
+    other = 4            // other
+  };
+
+  // data member defining a cell structure
+  class cell_id {
+  public:
+    using side_id = falaise::bounded_int<0,1>;
+    using layer_id = falaise::bounded_int<0,8>;
+    using column_id = falaise::bounded_int<0,112>;
+    using status_id = falaise::bounded_int<0,4>;
```

Comment:
I'd just use the `enum class` here as that should give the required type safety.

Reply:
Do you suggest having `*_id` as part of a `enum class`? In this case I am not sure I understand why. Thanks

Reply:
Sorry, I was rather rather unclear here! The "coordinates" would stay the same, but the `status_id` can be replaced with the `cell_status` `enum`, i.e.;

```c++
    using side_id = falaise::bounded_int<0,1>;
    using layer_id = falaise::bounded_int<0,8>;
    using column_id = falaise::bounded_int<0,112>;

...
 private:
     side_id side_;
     layer_id layer_;
     column_id column_;
     cell_status status_;
   };
```

So we can use the already defined `cell_status` rather than using another type for the same thing.